### PR TITLE
default Button type='button' - to avoid accidental trigger onEnter elsewhere

### PR DESCRIPTION
**Pull request by tarun**
## PR Checklist

- Manually tested across supported browsers
  - [x] Chrome
  - [x] Firefox
  - [x] Safari
  - [x] IE11 (Win 7)
  - [x] Edge (Win 10)
- [x] Unit tests written (`common` at minimum)
- [x] PR has one of the `semver-` labels
- [ ] Two core team engineer approvals
- [ ] One core team UX approval

…ing Enter key on the page

### CODE REVIEW

**Comment by tarun** [comment]
The issue - when hitting the enter key anywhere else on the page - text field, etc.
The onClick handler of the button is getting triggered... Ideally we want enter key to either submit the form or do nothing..
However Browsers default all buttons to type=submit by default - which wreaks havoc as hitting enter can trigger the onClick of any other button on the page  (technically the first in a form).

The generic use case of button is for form controls to modify UX behavior and other controls.. therefore this often leads to unexpected behavior.

By setting the default props as `<Button type=button />` - we avoid the accidental behavior in the 98% use case - and if a user wants to use a button to submit a form - they should explicitly specify `type=submit`
And this way we don't have to preventDefault on other form controls (text input) onEnter to avoid unexpected behavior.

File: src/components/Button/Button.jsx [hunk 1]
```diff
@@ -71,13 +71,18 @@ const Button = createClass({
 		 * Signature: `({ event, props }) => {}`
 		 */
 		onClick: func,
+		/**
+		 * form element type variations of button. Defaults to 'button' to avoid being triggered by 'Enter' anywhere on the page. Passed through to DOM button.
+		 */
+		type: string,
```
Comment (sodiumjoe) [inline src/components/Button/Button.jsx]: is this not redundant with `passThroughs`? i.e. doesn't this already work?

Comment (jondlm) [inline src/components/Button/Button.jsx]: It is redundant, but I think this makes it clear that we are defaulting it to `button` aye?

Comment (sodiumjoe) [inline src/components/Button/Button.jsx]: good pt

Comment (jondlm) [inline src/components/Button/Button.jsx]: Good question, not sure about that

**Pull request merged**